### PR TITLE
No physics until ready

### PR DESCRIPTION
**Pull request by sethalves**
- Don't simulate physics until enough of the scene is received over the network.

### CODE REVIEW

**Comment by sethalves** [comment]
https://app.asana.com/0/26225263936266/66264103181929/f

**Comment by hifi-gustavo** [comment]
Build console output for commit: c462bda86cb221aa7b8e5329f93367a42399d3d7 available at: https://gist.github.com/5824aa46a512fbd551f5<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6594.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6594.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2905,7 +2907,7 @@ void Application::update(float deltaTime) {
 
     _avatarUpdate->synchronousProcess();
 
-    {
+    if (_physicsEnabled) {
         PerformanceTimer perfTimer("physics");
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: Note: the avatar will be unable to move physically until **_physicsEnabled** is **true**.

What happens if a collision shape URL goes 404 or 500?  Will the object ever be ReadyToComputeShape?  Could a griefer set an invalid collision URL at the spawn point and thereby trap the physics of all incoming avatars?

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -3893,6 +3897,31 @@ void Application::trackIncomingOctreePacket(ReceivedMessage& message, SharedNode
...
+        entityTree->findEntities(box, entities);
+    });
+
+    foreach (EntityItemPointer entity, entities) {
+        if (!entity->isReadyToComputeShape()) {
+            return false;
+        }
+    }
+    return true;
+}
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: This is a step in the right direction, but is not foolproof.

Suppose we had a table with 100 BoxEntityItem's on top.  There is no guarantee that the entity-server will transmit the table first, so it is possible for the receiving interface to be in a state where there are 27 out of 100 boxes nearby that are all ready for physics, but the table data hasn't even been put on the wire yet.

A watertight solution to this problem is not easy.  Do we care?

Comment (sethalves) [inline interface/src/Application.cpp]: This code is only called from the code that processes stats packets.  Stats packets are only generated after a full scene send.

File: interface/src/Application.cpp (hunk 1 above)
Comment (sethalves) [inline interface/src/Application.cpp]: While physics is disabled, the old fallback avatar motion is available.  The user would have to move 3 meters away from the broken object before they could start simulating.

**Pull request merged**